connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -319,9 +319,18 @@ private void poll(long timeoutMs) {
         }
     }
 
-    private void readToLogEnd() {
+    /**
+     * This method finds the end offsets using the <code>listOffsets</code> method of the admin client.
+     * As the <code>listOffsets</code> method might throw a {@link RetriableException}, the <code>shouldRetry</code>
+     * flag enables retry, upon catching such exception, if it is set to <code>True</code>.
+     *
+     * @param shouldRetry Boolean flag to enable retry for the admin client <code>listOffsets</code> call.
+     */
+
+    private void readToLogEnd(boolean shouldRetry) {
         Set<TopicPartition> assignment = consumer.assignment();
-        Map<TopicPartition, Long> endOffsets = readEndOffsets(assignment);
+        // it will subsequently invoke the listOffsets call here
```

Comment:
I don't think this comment adds much value, especially because "subsequently" is ambiguous. IMO the method call stands on its own.
```suggestion
```